include/unifex/just.hpp

```diff
@@ -42,8 +44,14 @@ template <typename Receiver, typename... Values>
 struct _op<Receiver, Values...>::type {
   UNIFEX_NO_UNIQUE_ADDRESS std::tuple<Values...> values_;
   UNIFEX_NO_UNIQUE_ADDRESS Receiver receiver_;
+  bool done_;
 
   void start() & noexcept {
+    if(done_) {
```

Comment:
I'm not a fan of the change to make this a runtime decision about whether this holds a value or the one signal.

Better would be to build a `variant_sender<Senders...>` type that can hold a sender of one of several different types.

Reply:
Yes, indeed. Unfortunately each time we think we need another algorithm it's days of work to implement it. We can front load that work, or alternatively get something that works and tidy it later.

Reply:
Alternatively, as we know what we actually want to do for this, I could remove the just hack and have the cancellation always uncancel, with a comment and a TODO that we will fix with the variant sender later.